### PR TITLE
C3215 login to agent

### PR DESCRIPTION
**Pull request by DavidVallejosT**
Implement step definitions
Add logic to login with different user types

### CODE REVIEW

File: IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ namespace IDT.XGen.Tests.UI.StepDefinitions
     public class LoginSteps
```
Comment (mabalderramav) [inline IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs]: Add comment

File: IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@ namespace IDT.XGen.Tests.UI.StepDefinitions
...
     {
         private PortalWeb currentPortalWeb;
+        private UserType currentUserType;
```
Comment (mabalderramav) [inline IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs]: Change attribute name according naming convention. [see](https://github.com/coretech/br-bdd-automation/blob/master/Documentation/CodeStandardsAndNamingConvention.md)

File: IDT.XGen/Main/UI/Pages/Login/RetailerLogin.cs [hunk 3]
```diff
@@ -32,8 +33,33 @@ class RetailerLogin : BasePage, ILoginPage
         /// <param name="userType">User type in enum type.</param>
         public void LoginAsUserType(UserType userType)
```
Comment (mabalderramav) [inline IDT.XGen/Main/UI/Pages/Login/RetailerLogin.cs]: I think the logic of this method will improve. For Example
1.Create a class with the  "CredentialManager" name. It class should have the handles credentials.This class should have the responsability to handle the types of users.
2. You could use the structure of a JSON to store user types with their credentials.

> 
{
  "RetailerPortal": {
    "UserType1": {
      "name": "username",
      "password": "userpassword"
    },
    "UserType2": {
      "name": "username",
      "password": "userpassword"
    },
    "UserType3": {
      "name": "username",
      "password": "userpassword"
    }
  },
  "DistributorPortal": {
    "UserType1": {
      "name": "username",
      "password": "userpassword"
    },
    "UserType2": {
      "name": "username",
      "password": "userpassword"
    },
    "UserType3": {
      "name": "username",
      "password": "userpassword"
    }
  }
}

File: IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs [hunk 4]
```diff
@@ -15,6 +15,7 @@ namespace IDT.XGen.Tests.UI.StepDefinitions
     public class LoginSteps
     {
         private PortalWeb currentPortalWeb;
```
Comment (mabalderramav) [inline IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs]: Change attribute name according naming convention. [see](https://github.com/coretech/br-bdd-automation/blob/master/Documentation/CodeStandardsAndNamingConvention.md)

File: IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs (hunk 2 above)
Comment (DavidVallejosT) [reply IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs]: Done.

File: IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs (hunk 1 above)
Comment (DavidVallejosT) [reply IDT.XGen/Tests/UI/StepDefinitions/LoginSteps.cs]: Done.

File: IDT.XGen/Main/UI/Pages/Login/RetailerLogin.cs (hunk 3 above)
Comment (DavidVallejosT) [reply IDT.XGen/Main/UI/Pages/Login/RetailerLogin.cs]: I put a TODO in the method LoginAsUserType logic, for Not to be bloqued in the test case.
I will review JSON files in other task.

File: IDT.XGen/Main/UI/Pages/Login/RetailerLogin.cs [hunk 5]
```diff
@@ -6,6 +6,7 @@
 
 namespace IDT.XGen.Main.UI.Pages.Login
```
[the remark this answers is not in the document]
Comment (DavidVallejosT) [reply IDT.XGen/Main/UI/Pages/Login/RetailerLogin.cs]: Done.

**Pull request merged**